.github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml

```diff
@@ -0,0 +1,48 @@
+name: Build scylla-bench docker image with gocql PR
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled]
+
+jobs:
+  trigger-longevity-large-partition-asymmetric-cluster-3h-test:
+    if: contains(github.event.pull_request.labels.*.name, 'extended-ci')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Check out the scylla-bench repository
+        uses: actions/checkout@v2
+        with:
+          repository: scylladb/scylla-bench
+          path: scylla-bench
+
+      - name: Checkout GoCQL PR Repository
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+          path: gocql
+
+      - name: Build and push Scylla-bench Docker Image with gocql from PR
+        run: |
+          cd scylla-bench
+          GOCQL_REPO="github.com/${{ github.event.pull_request.head.repo.full_name }}" GOCQL_VERSION="${{ github.event.pull_request.head.ref }}" make build-with-custom-gocql-version
+          DOCKER_IMAGE_TAG="scylladb/gocql-extended-ci:scylla-bench-${{ github.event.pull_request.head.sha }}" DOCKER_IMAGE_LABELS="com.scylladb.gocql-version=${{ github.event.pull_request.head.sha }}" make build-sct-docker-image
+          docker push "scylladb/gocql-extended-ci:scylla-bench-${{ github.event.pull_request.head.sha }}"
+
+      - name: Start Jenkins job
+        uses: sylwiaszunejko/jenkins_client@main
+        with:
+          jenkins_job_name: 'scylla-staging/sylwia.szunejko/longevity-large-partition-asymmetric-cluster-3h-test'
+          jenkins_job_parameters: '{"email_recipients": "sylwia.szunejko@scylladb.com", "scylla_version": "6.1.1", "extra_environment_variables": "SCT_STRESS_IMAGE.scylla-bench=scylladb/gocql-extended-ci:scylla-bench-${{ github.event.pull_request.head.sha }}"}'
+          jenkins_base_url: 'https://jenkins.scylladb.com/'
+          jenkins_user: sylwiaszunejko
+          jenkins_password: ${{ secrets.JENKINS_TOKEN }}
+          wait_for_result: 'True'
+          polling_interval: '120'
+          polling_timeout: '12600'
```

Comment:
As far as I can tell, goland doesn't allow code execution during build, so it should not be possible to hijack CI using some malicious code.
The problem I see is that the result of such build is pushed to dockerhub and is available under scylladb org - are we ok with having potentially malicious images there?
Jenkins: the same problem. Is our infrastructure ready to execute untrusted code? I highly doubt it.

Reply:
it is pushed to a repo that is designated to this specific purpose, so chance that someone accidentially run this image is effectively zero.
This image is going to be run on a temporary instance which has access to the db nodes only, beside generic internet access, so you can consider it pretty safe.

Reply:
> Are you sure we are not putting some SSH keys / env variables etc on those nodes that should not be publicly available? Could someone from QA confirm it? @fruch @roydahan @soyacz

100% loader is getting controlled by sct-runner, which has all the keys, loaders them selves have nothing.

Reply:
1) those machines are not publicly available 
2) users we don't trust won't be able to execute this action, we'll need to actively add them as contributors to the project for this CI part to be working for them (as we do in lots of other public projects, i.e. scylla-core)

Reply:
So a maintainer will have to approve each CI run from outside user?

Reply:
> So a maintainer will have to approve each CI run from outside user?

Each external user, adding them as contributers.

maybe there's a way to configure the action to need approval on each run, but it's on the actions configurations

Reply:
> > So a maintainer will have to approve each CI run from outside user?
> 
> Each external user, adding them as contributers.

After being added as contributors they wouldn't need approval anymore? In that case account overtake of such user would be dangerous for us.

> 
> maybe there's a way to configure the action to need approval on each run, but it's on the actions configurations

It is possible to require CI approval for all runs from external contributors:
![image](https://github.com/user-attachments/assets/db8a9b39-f55e-46af-9907-e66237f5472e)

I don't know any way to require this for a single job.

Reply:
There is a way via deployment environment, but us realy inconvinient, since code is ran in isolated environment I don't see any reason on doing that.